### PR TITLE
Nbo/download logs prod 7515

### PR DESCRIPTION
**Pull request by Nikeleos**

### CODE REVIEW

File: src/components/ScenarioParameters/ScenarioParameters.js [hunk 1]
```diff
@@ -76,6 +76,9 @@ const ScenarioParameters = ({
     const initialStockParameter = currentScenario.data?.parametersValues?.find(el => el.parameterId === INITIAL_STOCK_PARAM_ID);
     setInitialStockDatasetId(initialStockParameter?.value);
     /// //////////////////////////////////////////////////////////////////////// INITIAL STOCK
+
+    // Reset parameters
+    resetParameters();
```
Comment (csm-thu) [inline src/components/ScenarioParameters/ScenarioParameters.js:81]: This change is not a part of your branch, could you rebase your branch on **main** please ? :)

**Review by jreynard-code — CHANGES_REQUESTED** [review]
Few changes to make but GG ;)

File: src/components/ScenarioParameters/ScenarioParameters.js (hunk 1 above)
Comment (Nikeleos) [reply src/components/ScenarioParameters/ScenarioParameters.js]: Resovled by rebase on main

File: src/views/Scenario/Scenario.js [hunk 2]
```diff
@@ -133,6 +134,10 @@ const Scenario = (props) => {
               scenarioId={currentScenario.data.id}
               scenarioState={currentScenario.data.state}
               csmSimulationRun={currentScenario.data.lastRun?.csmSimulationRun}
+              downloadCumulatedLogs={() => {
```
[the remark this answers is not in the document]
Comment (Nikeleos) [reply src/views/Scenario/Scenario.js:137]: Code refactored

File: src/utils/ScenarioRunUtils.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import ScenariorunService from '../services/scenariorun/ScenariorunService.js';
+import fileDownload from 'js-file-download';
+import { ORGANISATION_ID } from '../configs/App.config';
+
+export const LOG_TYPES = {
+  SIMPLE_LOGS: 'SIMPLE_LOGS',
+  CUMULATED_LOGS: 'CUMULATED_LOGS'
+};
+
+const downloadScenarioRunCumulatedLogsFile = (lastRun) => {
```
[the remark this answers is not in the document]
Comment (Nikeleos) [reply src/utils/ScenarioRunUtils.js:13]: Done

File: src/utils/ScenarioRunUtils.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  const logsPromise = ScenariorunService.getScenarioRunCumulatedLogs(ORGANISATION_ID, lastRun.scenarioRunId);
+  logsPromise.then(result => {
+    const blob = new Blob([result.data]);
+    fileDownload(blob, 'cumulated_logs.txt');
+  });
+};
+
+const downloadScenarioRunLogsFile = (lastRun) => {
```
[the remark this answers is not in the document]
Comment (Nikeleos) [reply src/utils/ScenarioRunUtils.js:21]: Done

File: src/views/Scenario/Scenario.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nikeleos) [reply src/views/Scenario/Scenario.js]: Done

File: src/utils/ScenarioRunUtils.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (Nikeleos) [reply src/utils/ScenarioRunUtils.js]: Done

File: src/utils/ScenarioRunUtils.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Nikeleos) [reply src/utils/ScenarioRunUtils.js]: Done

File: src/views/Scenario/Scenario.js [hunk 5]
```diff
@@ -140,6 +142,13 @@ const Scenario = (props) => {
               scenarioName={currentScenario.data?.name}
               scenarioState={currentScenario.data?.state}
               noScenario={noScenario}
+              scenarioId={currentScenario.data?.id}
+              downloadLogsFile={() => {
+                ScenariorunService.downloadLogsFile(
+                  currentScenario.data.lastRun,
```
Comment (csm-thu) [inline src/views/Scenario/Scenario.js:148]: Not sure if this is necessary, but `currentScenario.data?.lastRun` (with a question mark) would be more consistent with the lines above

File: src/state/reducers/scenario/ScenarioReducer.js [hunk 6]
```diff
@@ -20,11 +20,11 @@ export const scenarioListReducer = createReducer(scenarioListInitialState, (buil
       state.data = action.list;
       state.status = STATUSES.SUCCESS;
     })
-    .addCase(SCENARIO_ACTIONS_KEY.UPDATE_SCENARIO_STATE, (state, action) => {
+    .addCase(SCENARIO_ACTIONS_KEY.UPDATE_SCENARIO, (state, action) => {
       state.data = state.data.map(scenarioData => {
-        // Replace state in data for the scenario to update
+        // Replace state and lastun in data for the scenario to update
```
Comment (csm-thu) [inline src/state/reducers/scenario/ScenarioReducer.js:25]: Typo: lastun -> lastRun

File: src/state/reducers/scenario/ScenarioReducer.js [hunk 7]
```diff
@@ -20,11 +20,11 @@ export const scenarioListReducer = createReducer(scenarioListInitialState, (buil
...
         if (scenarioData.id === action.data.scenarioId) {
-          return { ...scenarioData, state: action.data.scenarioState };
+          return { ...scenarioData, state: action.data.scenarioState, lastRun: action.data.lastrun };
```
Comment (csm-thu) [inline src/state/reducers/scenario/ScenarioReducer.js:27]: Shouldn"t it be lastRun instead of lastrun ?

Comment (Nikeleos) [reply src/state/reducers/scenario/ScenarioReducer.js:27]: Fixed all occurences of scenarioRun case

File: src/state/reducers/scenario/ScenarioReducer.js (hunk 6 above)
Comment (Nikeleos) [reply src/state/reducers/scenario/ScenarioReducer.js:25]: Fixed

Comment (Nikeleos) [reply src/state/reducers/scenario/ScenarioReducer.js:25]: Fixed

File: src/views/Scenario/Scenario.js (hunk 5 above)
Comment (Nikeleos) [reply src/views/Scenario/Scenario.js:148]: Right. Fixed

Comment (Nikeleos) [reply src/views/Scenario/Scenario.js]: Fixed

File: src/state/reducers/scenario/ScenarioReducer.js (hunk 6 above)
Comment (Nikeleos) [reply src/state/reducers/scenario/ScenarioReducer.js:25]: Fixed

File: src/state/reducers/scenario/ScenarioReducer.js (hunk 7 above)
Comment (Nikeleos) [reply src/state/reducers/scenario/ScenarioReducer.js]: Fixed

**Pull request merged**